src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.NodeMapBuilder.cs

```diff
@@ -255,6 +255,19 @@ public override BoundNode VisitBinaryOperator(BoundBinaryOperator node)
                 throw ExceptionUtilities.Unreachable;
             }
 
+            public override BoundNode VisitDeconstructionAssignmentOperator(BoundDeconstructionAssignmentOperator node)
+            {
+                // For deconstruction declarations, the BoundLocals in the LeftVariables should not be added to the map
+                if (!node.IsDeclaration)
+                {
+                    VisitList(node.LeftVariables);
+                }
+
+                Visit(node.Right);
+                // don't map the deconstruction, conversion or assignment steps
+                return null;
+            }
```

Comment:
The most direct test is via `GetTypeInfo`. Without the above changes to avoid the visitor putting too many nodes in the map, one cannot get the type info for `(1, 2)` in `var (x, y) = (1, 2);`.